__tests__/nc-news6.test.js

```diff
@@ -0,0 +1,53 @@
+const request = require("supertest");
+const app = require("../app.js");
+const db = require("../db/connection.js");
+const { seed } = require("../db/seeds/seed.js");
+const data = require("../db/data/test-data/topics.js");
+
+afterAll(() => {
+  return db.end;
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("The array should return 200 all ok", () => {
+    return request(app).get("/api/articles/3/comments").expect(200);
+  });
+  test("should return 400 if the article_id is a string or a combination of number and string", () => {
+    return request(app).get("/api/articles/banana1/comments").expect(400);
+  });
+  test("should return 404 if an invalid id is provided", () => {
+    return request(app)
+      .get("/api/articles/99")
+      .expect(404)
+      .catch(({ body }) => {
+        expect(body.message).toBe("Id Not found");
+      });
+  });
+  test("should return an array of comments for the article id provided", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments[3]).toMatchObject({
```

Comment:
This is an improvement however we should do this assertion for every object in the array. Can you think of a way to this forEach of the objects 😉 ? If you can you MUST first assert the length of the array is correct before doing so. Please resubmit when you have done this. 👍